sklearn/utils/tests/test_param_validation.py

```diff
@@ -571,3 +579,11 @@ class SomeType:
 
     f(param2=SomeType)
     f(param2=SomeType())
+
+
+def test_missing_values_pd_na():
+    """Add a specific test for checking support for `pandas.NA`."""
+    pd = pytest.importorskip("pandas")
+
+    missing_values_constraint = _MissingValuesHelper()
```

Comment:
I'd rather test _NAConstraint, and also check that np.nan does not satisfy it (and maybe also that an array does not satisfy it)